bundles/org.openhab.voice.actiontemplatehli/src/main/java/org/openhab/voice/actiontemplatehli/internal/ActionTemplateInterpreter.java

```diff
@@ -0,0 +1,1159 @@
+        String tag = null;
+        if (placeholderConfig.posFile != null) {
+            File posTaggingDictionary = Path.of(POS_FOLDER, placeholderConfig.posFile + ".xml").toFile();
+            File posTaggingModel = Path.of(POS_FOLDER, placeholderConfig.posFile + ".bin").toFile();
+            if (posTaggingModel.exists()) {
+                POSModel posModel = new POSModel(posTaggingModel);
+                var tags = new POSTaggerME(posModel).tag(new String[] { singleWorldText });
+                if (tags.length > 0 && !"O".equals(tags[0])) {
+                    tag = tags[0];
+                }
+            } else if (posTaggingDictionary.exists()) {
+                POSDictionary posDictionary = POSDictionary.create(new FileInputStream(posTaggingDictionary));
+                var tags = posDictionary.getTags(singleWorldText);
+                if (tags != null && tags.length > 0 && !"O".equals(tags[0])) {
+                    tag = tags[0];
+                }
+            } else {
+                logger.warn("configured pos transformation file not found {}", placeholderConfig.posFile);
+            }
+        } else if (placeholderConfig.posStaticValues != null) {
+            var dictionary = new POSDictionary(config.caseSensitive);
+            for (var entry : placeholderConfig.posStaticValues.entrySet()) {
+                dictionary.put(entry.getKey(), entry.getValue());
+            }
+            var tokenTags = dictionary.getTags(singleWorldText);
+            if (tokenTags != null && tokenTags.length > 0) {
+                tag = tokenTags[0];
+            }
+        } else {
+            // no transformation configured
+            return text;
+        }
+        if (tag == null) {
+            return "";
+        }
+        return tag.replaceAll("__", " ");
+    }
+
+    private Map<String[], Item> getItemsByLabelTokensMap() throws IOException {
+        var maps = this.nlpItemMaps;
+        if (maps == null) {
+            return getItemsMaps().itemLabelByTokens;
+        }
+        return maps.itemLabelByTokens;
+    }
+
+    private Map<Item, ActionTemplateConfiguration[]> getItemsWithActionConfigs() throws IOException {
+        var maps = this.nlpItemMaps;
+        if (maps == null) {
+            return getItemsMaps().itemsWithActionConfigs;
+        }
+        return maps.itemsWithActionConfigs;
+    }
+
+    private NLPItemMaps getItemsMaps() throws IOException {
+        var itemMaps = this.nlpItemMaps;
+        if (itemMaps == null) {
+            var itemByLabelTokens = new HashMap<String[], Item>();
+            var itemsWithActionConfigs = new HashMap<Item, ActionTemplateConfiguration[]>();
+            for (Item item : itemRegistry.getAll()) {
+                var label = item.getLabel();
+                if (label != null) {
+                    var lowerLabel = label.toLowerCase();
+                    itemByLabelTokens.put(tokenizeText(lowerLabel), item);
```

Comment:
In case several items have the same label, you keep trace of only one.
This is not a problem ?

Reply:
Yes, there is no support for that. I will a warning in case that happens so the user is aware, thanks for pointing it.

Reply:
I've added a warning. But maybe this situation is common (it's not my case), should it rather be a warning?

Reply:
I still not happy with this, I'll add support for item 'Synonyms' and just show a debug log for the collisions.